enrolments/validators.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from django.core.exceptions import ValidationError
+from registration.models import Student
+
+
+def validate_attendance(classObj):
+    """
+    Validates against the attendance column for the Class model.
+    The expected structure is:
+    [
+        {
+            "date": "2021-04-19",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        },
+        {
+            "date": "2021-04-26",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        }
+    ]
+    """
+    for session in classObj:
+        date = session["date"]
+        attendees = session["attendees"]
+        try:
+            datetime.strptime(date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValidationError(
+                "%(date) must be formatted as YYYY-MM-DD and must be a valid date",
+                code="invalid",
+                params={"date": date},
+            )
+        if attendees and not Student.objects.filter(pk__in=attendees).exists():
```

Comment:
i believe this validates if at least one ID matches a student, even if the rest don't exist – should we instead raise an error if _any_ ID is invalid?